### PR TITLE
Booleans are not parsed as functions. Fixes #74.

### LINKED ISSUE #74: Booleans getting matched as .entity.name.function
```coffee
@test: (a) ->
  return false unless true
#        ^^^^^ this is being marked as `entity.name.function`.  The `true` is correct though.
```

language-coffee-script@0.45.0

### PR DESCRIPTION
**Pull request by jonboiser**
Added specs based on bug in #74. Seems to fix #74. Also includes diffs from PR #76.

### CODE REVIEW

**Comment by 50Wliu** [comment]
I just merged #76 - can you please rebase this on top of master so that only the diffs from this PR show up?  Thanks.

**Comment by 50Wliu** [comment]
Also, can we just move the boolean matching above the function pattern?

**Comment by jonboiser** [comment]
Sure. I moved the boolean matchers right before the function ones, and reverted the change I made to the function pattern. New specs still pass.

File: grammars/coffeescript.cson [hunk 1]
```diff
@@ -245,6 +237,15 @@
     'name': 'constant.language.boolean.false.coffee'
   }
   {
+    'match': '@?\\b(?!class|subclass|extends|decodeURI(Component)?|encodeURI(Component)?|eval|parse(Float|Int)|require)\\w+(\\s+(?!(of|in|then|is|isnt|and|or|for|else|when|not|if)\\s)(?=(@?\\w+|\\->|\\-\\d|\\[|\\{|\"|\'))|(?=\\())'
+
```
Comment (50Wliu) [inline grammars/coffeescript.cson]: Extra newline here

**Comment by jonboiser** [comment]
Cleaned up the commit history and removed that newline.

**Comment by 50Wliu** [comment]
Thanks!

**Pull request merged**